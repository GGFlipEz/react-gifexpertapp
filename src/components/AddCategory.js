import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ( { setCategories } ) => {
    
    //const [inputValue, setInputValue] = useState(); --> Resulta Undefied
    const [inputValue, setInputValue] = useState(''); //Regresa Vacio, no produce error.

    const handleInputChange = (e) => {
        //console.log(e.target.value);
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2 ){
            setCategories( cats => [ inputValue, ...cats]);
            setInputValue('');
        }

    }

    return (
            <form onSubmit={ handleSubmit}>
            <h2>AddCategory</h2>
                <input
                    type="text"
                    value={ inputValue }
                    onChange={handleInputChange}
                />
            </form>
    )
}


AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}