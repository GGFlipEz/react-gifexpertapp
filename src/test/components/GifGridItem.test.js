import React  from "react";
import { GifGridItem } from "../../components/GifGridItem";
import { shallow, configure } from 'enzyme';



describe ('Pruebas en <GifGridItem />', () => {

    const title = 'Test';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title= {title} url= {url} /> )

    test('debe mostrar el componente correctamente ', () => {

        
        expect( wrapper ).toMatchSnapshot();

    })


    test('debe tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })

    test('debe de tener la imagen igual al url y alt de los props ', () => {
        
        const  img = wrapper.find('img');
        // console.log( img.props() ) ;
        // console.log( img.props().src ) ;
        // console.log( img.props().alt ) ;
        expect( img.props().src ).toBe( url );
        expect( img.props().alt ).toBe( title );
        

    })

    test('Debe tener animate_fadeIn', () => {
        const div = wrapper.find('div');
        //animate_fadeIn
        //console.log( div.props('className') );
        const className = div.props().className;
        expect( className.includes('animate_fadeIn') ).toBe( true );


    })



})