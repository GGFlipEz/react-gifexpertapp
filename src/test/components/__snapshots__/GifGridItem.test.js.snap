// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente  1`] = `
<div
  className="card animate_animated animate_fadeIn"
>
  <img
    alt="Test"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Test
     
  </p>
</div>
`;
